web_console/components/Form.jsx

```diff
@@ -218,6 +289,87 @@ export default function Form({
     }
   };
 
+  const renderFieldInGrid = x => (<Grid key={x.key} xs={x.span || 8} md={x.span || 8}>{renderField(x)}</Grid>)
+
+  const renderGroup = group => {
+    // const [, setToast] = useToasts()
+    const [formType, setFormType] = useState(group.formTypes && group.formTypes[0])
+    const [groupFields, setGroupFields] = useState(
+      Array.isArray(group.fields) ? group.fields : group.fields[formType]
+    )
+
+    /**
+     * form type switch
+     * - set `formTypes` to active form type switch
+     * - fileds must be object with key as form type, value as fields of group
+     * - set `onFormTypeChange` to convert data between types
+     *    - (data, currType, targetType) => object
+     *    - data is an object with form data (json fields as string)
+     *    - the returned object will be used to fill form (json needs to be string)
+     *    - if `error` field in the returned object, switching will be prevented and show error msg
+     */
+    const onFormTypeChange = (e, type) => {
+      e.stopPropagation()
+
+      if (formType === type) return
+      // get form data
+      let data = groupFields.reduce((total, curr) => {
+        total[curr.key] = form[curr.key]
+        return total
+      }, {})
+      let res = group.onFormTypeChange && group.onFormTypeChange(data, formType, type)
+      if (res.error) {
+        // TODO: figure out why this not working
+        // setToast({ text: res.error, type: 'error' })
+        alert(res.error)
+        return
+      }
+
+      let copy = {...form}
+      for (let key in res) {
+        copy[key] = res[key]
+      }
+      setForm(copy)
+
+      setFormType(type)
+      setGroupFields(group.fields[type])
+    }
+    const collapseTitle = () => {
+      return <>
+        {group.groupName}
+        { formType ?
+          <ButtonGroup className="formTypeBtns" size="small" style={{marginLeft: '20px'}}>
+            { group.formTypes.map(type =>
+                <Button
+                  className={formType === type ? 'selecetedType' : 'formTypeBtn'}
+                  onClick={e => onFormTypeChange(e, type)}
+                >
+                  {type}
+                </Button>
+              ) }
+            <style jsx global>{`
```

Comment:
it'd be better to use `styled-jsx` for css generation just like what `useStyles` do above